content/guides/server-sent-events.adoc

```diff
@@ -10,26 +10,164 @@ toc::[]
+build a template project. If you haven't already installed it, please
+take a few minutes to set it up.
+
+== Generate Our New Project
+
+----
+$ lein new pedestal-service myapp
+Generating a pedestal-service application called server-sent-events.
+$ cd server-sent-events
+----
+
+== Add Hiccup
+
+We will be using Hiccup to generate our index page, so we will need to add the dependency in our project.clj.
+
+[source,clojure]
+.project.clj
+----
+include::server-sent-events/project.clj[tags=add-hiccup]
+----
+
+== Clean Up
+
+Here's the service.clj file generated for us:
+
+[source,clojure]
+.src/server_sent_events/service.clj
+----
+include::server-sent-events/src/server_sent_events/service.clj[tags=original-service]
+----
+
+Let's simply things by uncommenting the 'Terse/Vector-based routes' section and removing uneccessary code.
+
+[source,clojure]
+.src/server_sent_events/service.clj
+----
+include::server-sent-events/src/server_sent_events/service.clj[tags=cleaned-up]
+----
+
+== Server-Sent Event Route
+
+Now let's add our server-sent event route. First Lets require SSE and core.async.
+
+[source,clojure]
+.src/server_sent_events/service.clj
+----
+include::server-sent-events/src/server_sent_events/service.clj[tags=require-sse]
+----
+
+Now we'll add our counter route. We create the "/counter" route by calling sse/start-event-stream and passing in our stream-ready handler. This creates an interceptor that will handle setting up the connection and call the stream-ready handler. A stream-ready handler receives a core.async event-chan and SSE context as parameters. You send events on the channel using regular core.async (such as async/>!!) functions and terminate the connection by calling async/close! on the channel. Here our event consists the :name and :data keys, but could also contain an :id for the event.
+
+[source,clojure]
+.src/server_sent_events/service.clj
+----
+include::server-sent-events/src/server_sent_events/service.clj[tags=counter-route]
+----
+
+Now lets start up our server.
+
+----
+$ lein run
+...
+Creating your server...
+...
+INFO  org.eclipse.jetty.server.Server - Started @4434ms
+----
+
+And checkout out our event stream in another terminal using curl.
+
+----
+$ curl http://localhost:8080/counter
+event: counter
+data: 0
+
+event: counter
+data: 1
+
+event: counter
+data: 2
+
+event: counter
+data: 3
+
+event: counter
+data: 4
+
+event: counter
+data: 5
+
+event: counter
+data: 6
+
+event: counter
+data: 7
+
+event: counter
+data: 8
+
+event: counter
+data: 9
+
+----
+
+Now lets stop our server (CTRL-C) and update our index route to return an HTML page that connects to our counter route. Since we are adding inline JS we will need to also update our service config to not include secure headers (which include 'unsafe-inline' by default).
+
+[source,clojure]
+.src/server_sent_events/service.clj
+----
+include::server-sent-events/src/server_sent_events/service.clj[tags=html-home]
```

Comment:
This section feels rushed. JS is the more important client. Maybe link out to a 3rd party tutorial on the client-side part of SSE? (Since it's not Pedestal-specific, it seems like we don't need to cover it in detail here.)